src/gravel/controllers/nodes/mgr.py

```diff
@@ -414,35 +414,35 @@ async def _spawn_etcd(
 
         logger.info(f"starting etcd, hostname: {hostname}, addr: {address}")
 
-        def _get_etcd_args() -> str:
+        def _get_etcd_args() -> List[str]:
             client_url: str = f"http://{address}:2379"
             peer_url: str = f"http://{address}:2380"
 
             nonlocal initial_cluster
             if not initial_cluster:
                 initial_cluster = f"{hostname}={peer_url}"
 
-            args_dict: Dict[str, str] = {
-                "name": hostname,
-                "initial-advertise-peer-urls": peer_url,
-                "listen-peer-urls": peer_url,
-                "listen-client-urls": f"{client_url},http://127.0.0.1:2379",
-                "advertise-client-urls": client_url,
-                "initial-cluster": initial_cluster,
-                "initial-cluster-state": "existing",
-                "data-dir": f"/var/lib/etcd/{hostname}.etcd"
-            }
+            args: List[str] = [
+                "--name", hostname,
+                "--initial-advertise-peer-urls", peer_url,
+                "--listen-peer-urls", peer_url,
+                "--listen-client-urls", f"{client_url},http://127.0.0.1:2379",
+                "--advertise-client-urls", client_url,
+                "--initial-cluster", initial_cluster,
+                "--initial-cluster-state", "existing",
+                "--data-dir", f"/var/lib/etcd/{hostname}.etcd"
+            ]
 
             if new:
                 assert token
-                args_dict["initial-cluster-token"] = token
-                args_dict["initial-cluster-state"] = "new"
+                args += ["--initial-cluster-state", "new"]
```

Comment:
I think this means args will contain both `--initial-cluster-state existing` _and_ `--initial-cluster-state new` in the new case. Is that going to be a problem?

Reply:
Right, this might or might not work depending on how etcd deals with overriding option values. I'd rather do this right on our end instead of hoping etcd does what we think should happen.

Reply:
oops, good catch! hanks @tserong